lists-service/cicd/base/app/base.yaml

```diff
@@ -89,4 +128,21 @@ Outputs:
 
   ListsSecret:
     Description: The Secrets name for lists 
-    Value: !Ref ListsSecret
\ No newline at end of file
+    Value: !Sub
+      - lists-${ResourceName}
+      - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
```

Comment:
I intentionally used the !Ref ListsSecret so the name wouldn't have to be constructed twice, it's a bit of a gotcha if the secrets name is changed, it had to be changed identically here as well. The downside is the secrets name has to parsed from the URI but i thought that was the lesser of the evils

Reply:
That's a good point. 
The reason I changed it was so I could use `!ImportValue` to set the environment for the deploy code build action (without needing to processing the ARN). 
It might be possible to use a convoluted combination if `!Split`, `!Select` to strip out the secret name from the ARN.

Unfortunately there isn't the ability to create a re-usable variable, there is a feature request for this https://github.com/aws-cloudformation/cfn-language-discussion/issues/55